File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3

import rclpy 
from rclpy.node import Node
from example_interfaces.msg import String 


class RobotNewsStation (Node) :
    def __init__(self) :
        super().__init__("robot_news_station")
        # Creating a Publisher 
        # Need to provide name, data type and queue size
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        # this will publish the message very 0.5 seconds
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station node has been started.")

    def publish_news (self) :
        # Creating the message 
        msg = String()
        msg.data = "Hello"
        # publishing the message 
        self.publisher_.publish(msg)
        






def main (args=None) :
    
    rclpy.init(args=args)
    node = RobotNewsStation()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__" :
    main()

File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class RobotNewsStationNode : public rclcpp::Node {

    public:
    RobotNewsStationNode () : Node("robot_news") {
        // Creating the publisher 
        // create publisher method takes the data type of the msg , topic name and queue size
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(0.5s, [this] () {publish_news();});

        RCLCPP_INFO(this->get_logger(), "Robot News Station Node started");

    }
    

    private:
    void publish_news () {
        auto msg {example_interfaces::msg::String()};
        msg.data = "Hi, from the c++ publisher";
        publisher_->publish(msg);
    }
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;


    
};


int main (int argc, char **argv) {

    rclcpp::init(argc, argv);


    auto node {std::make_shared<RobotNewsStationNode>()};

    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}